capplets/time-admin/src/time-tool.c

```diff
@@ -0,0 +1,317 @@
+/*  time-admin 
+*   Copyright (C) 2018  zhuyaliang https://github.com/zhuyaliang/
+*
+*   This program is free software: you can redistribute it and/or modify
+*   it under the terms of the GNU General Public License as published by
+*   the Free Software Foundation, either version 3 of the License, or
+*   (at your option) any later version.
+
+*   This program is distributed in the hope that it will be useful,
+*   but WITHOUT ANY WARRANTY; without even the implied warranty of
+*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*   GNU General Public License for more details.
+
+*   You should have received a copy of the GNU General Public License
+*   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include "time-tool.h"
+
+struct tm *GetCurrentTime(void)
```

Comment:
Remove `struct tm *GetCurrentTime(void)` and use [GDateTime *g_date_time_new_now_local (void);](https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-new-now-local) and adapt the code.

Reply:
Why not use ```struct tm```

Reply:
[GDateTime](https://developer.gnome.org/glib/stable/glib-GDateTime.html) is based on object-oriented programming and it has useful functions, like g_date_time_get_year (GDateTime *datetime) that doesn't require to add a base year (1900).

Reply:
Thanks for your help If  ```struct tm``` has no other big problems, I don't think it's necessary to replace it.